llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -8291,7 +8291,39 @@ SDValue AArch64TargetLowering::LowerFormalArguments(
   if (Subtarget->hasCustomCallingConv())
     Subtarget->getRegisterInfo()->UpdateCustomCalleeSavedRegs(MF);
 
-  if (!getTM().useNewSMEABILowering() || Attrs.hasAgnosticZAInterface()) {
+  if (getTM().useNewSMEABILowering() && !Attrs.hasAgnosticZAInterface()) {
+    if (Subtarget->isTargetWindows() || hasInlineStackProbe(MF)) {
+      SDValue Size;
+      if (Attrs.hasZAState()) {
+        SDValue SVL = DAG.getNode(AArch64ISD::RDSVL, DL, MVT::i64,
+                                  DAG.getConstant(1, DL, MVT::i32));
+        Size = DAG.getNode(ISD::MUL, DL, MVT::i64, SVL, SVL);
+      } else if (Attrs.hasAgnosticZAInterface()) {
```

Comment:
`Attrs.hasAgnosticZAInterface()` is always false, because the condition above `if (getTM().useNewSMEABILowering() && !Attrs.hasAgnosticZAInterface())`.

Reply:
I'm missing a test for agnostic ZA with stack problems then  :sweat_smile:. The `!Attrs.hasAgnosticZAInterface()` above should be removed.

Reply:
Oh no, the agnostic-ZA changes should not be added until the next PR (this is an artifact of how I broke up the patches).